common.py

```diff
@@ -524,3 +529,71 @@ def build_sriov_network_operator_check_permissions() -> bool:
     rsh = host.LocalHost()
     ret = rsh.run("podman pull registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.21-openshift-4.16")
     return ret.success()
+
+
+def git_repo_setup(repo_dir: str, *, repo_wipe: bool = True, url: str, branch: str = "master") -> None:
+    exists = os.path.exists(repo_dir)
+    if exists and not repo_wipe:
+        return
+
+    if exists:
+        shutil.rmtree(repo_dir)
+
+    logger.info(f"Cloning repo {url} to {repo_dir}")
+    Repo.clone_from(url, repo_dir, branch=branch)
+
+
+@dataclasses.dataclass
+class ContainerInfo:
+    name: str
+    envvar: str
+    containerfile: str
+    registry: str
+    project: str
+    full_tag: str = dataclasses.field(init=False)
+
+    def __post_init__(self) -> None:
+        self.full_tag = f"{self.registry}/{self.project}/{self.name}:latest"
+
+
+def start_image_registry(rsh: host.Host, client: K8sClient) -> str:
+    reglocal_dir_name, reglocal_hostname, reglocal_listen_port, reglocal_id = reglocal.ensure_running(rsh)
+
+    reglocal.ocp_trust(client, reglocal_dir_name, reglocal_hostname, reglocal_listen_port)
+
+    registry = f"{reglocal_hostname}:{reglocal_listen_port}"
+
+    return registry
+
+
+def git_build_local(rsh: host.Host, repo_dir: str, registry: str, project: str, container_infos: list[ContainerInfo]) -> dict[str, str]:
+    for ci in container_infos:
+        if os.environ.get("CDA_LOCAL_IMAGE_REBUILD") == "0" and rsh.run(shlex.join(["podman", "images", "-q", ci.full_tag])).out:
```

Comment:
is this environmental variable documented somewhere?

Reply:
This is not documented, and should probably be added in a follow up PR for the reglocal module. This is outside the scope of this PR, this is just a refactor of some existing code.